### ISSUE TITLE
Create entry points for compiling `.lissp` files

### ISSUE THREAD

**Issue by brandonwillard** [gilch/hissp#24]
We should probably set up an `argparse` interface (with help strings) to the existing entry point, `hissp`, that starts the REPL, compiles files, etc.

**Comment by gilch** [comment]
I wonder which options we actually want? You only mentioned the REPL and compiling files. I want Hissp to have very limited scope (only the basics) with extension projects living in separate repositories. A less-basic REPL would fall into that category.

Currently, `hissp` only starts the basic Lissp REPL. As explained in the FAQ, you can invoke the `hissp.reader..transpile` function to compile a `.lissp` file, from the REPL. This extra step may seem annoying, but once you have a REPL open, it's no more difficult to type a command into the REPL than into a terminal. This means that many other command-line options could be implemented as REPL functions.

But my current recommendation is to have the `__init__.py` automatically compile any `.lissp` files in the package. The `transpile` function is designed to make that easy.

For comparison, `python` also starts its interactive shell, but compiles everything automatically.

What about other options? Python can also launch a python file or module (with `-m`) or drop into an interactive session after running a module (with `-i`). It can also execute a command passed directly (with `-c`), or read them from stdin (with `-`). I can see how these, or something similar, might be useful for Hissp, but I am not sure how they should work. I certainly wished I had something like `-i` when I was writing the basic macros. Python also has many other options that seem less relevant, but if we want to allow `hissp` to be a replacement for `python`, then perhaps we have to pass them through somehow. I see a lot of potential for bloat here.

With Hissp's qualified symbols and macros and explicit `()`, I see the potential for it being good at one-liners for use in shell pipelines, unlike Python, but perhaps more like Perl. So perhaps some of the options Perl has, but Python lacks are worth considering.

I was also considering an option to change which macros the REPL starts with, but perhaps this overlaps with `-i`. On the other hand, Python has both the `-i` and the `PYTHONSTARTUP` environment variable.

**Comment by gilch** [comment]
I added a lisspc. It probably still needs work. Command line stuff is not tested well enough.

**Comment by gilch** [comment]
#62 redoes the basic argparse interface. We've got a `-i` and a `-c`. It can run the new REPL or a file or string as main, and optionally drop into the REPL after that.

It also removed the `lisspc` entry point from `setup.py` for now. And (for now) you can still invoke its former backing entry point by running the reader module as main, like `python -m hissp.reader "" spam`.

`lisspc` never quite worked right. Lissp was just not designed to be compiled from the command line like that. It's better to do `transpile()` in `__init__.py`'s and in main. But I can still see a potential need for compiling main in the first place, even though you can run it directly, there are use cases where you'd want it compiled to Python first. My original intent was to do that from the REPL, which does work.

I can also see the use of a build system completely external to Python in a large project. It would be nice to be able to push a button in your IDE and do a clean build of everything. This kind of thing may seem easier with a `lisspc`, but I'm not sure if it actually would be. You can't just compile a file in isolation. It needs to know its package.

You can run a one-off Lissp command with `lissp -c` now, including (say) `$ lissp -c "(hissp.reader..transpile None foo)"` or even `$ lissp -c "(hissp.reader..transpile : :* (getitem sys..argv (slice 1 None))" "" foo`, which should be sufficient to implement external build systems.

**Comment by gilch** [comment]
Removed the main entry point from `hissp.reader` to get full test coverage. I also documented the `lissp -c` workaround, and suggested using aliases. Given these, and the difficulty of testing shell commands, I'm starting to think a command-line entry point is not worth having in Hissp proper.

A command-line interface could still be an external tool. There are a number of Python libraries that make CLIs easier to implement, but Hissp is not supposed to have dependencies.
